Sources/PointFree/Subscribe.swift

```diff
@@ -8,28 +8,14 @@ import Prelude
 import Stripe
 import Tuple
 
-let subscribeMiddleware =
-  filterMap(
-    require1 >>> pure,
-    or: redirect(
-      with: get1 >>> subscribeConfirmationWithSubscribeData,
-      headersMiddleware: flash(.error, "Error creating subscription!")
-    )
-    )
-    <<< filter(
-      get1 >>> ^\.pricing >>> validateQuantity,
-      or: redirect(
-        with: get1 >>> subscribeConfirmationWithSubscribeData,
-        headersMiddleware: flash(.error, "An invalid subscription quantity was used.")
-      )
-    )
-    <<< filter(
-      get1 >>> validateCoupon(forSubscribeData:),
-      or: redirect(
-        with: get1 >>> subscribeConfirmationWithSubscribeData,
-        headersMiddleware: flash(.error, "Coupons can only be used on individual subscription plans.")
-      )
-    )
+let subscribeMiddleware: Middleware<
+  StatusLineOpen,
+  ResponseEnded,
+  Tuple2<SubscribeData?, User?>,
+  Data
+  > = requireSubscribeData
```

Comment:
this chain of middlewares was segfaulting.

someday we should try what you did for reducers on middleware. middleware is probably a lot more similar to reducers than views.

Reply:
Yeah that sounds good.